apps/tai/test/tai/venue_adapters/okex/stream/trades_test.exs

```diff
@@ -10,25 +10,30 @@ defmodule Tai.VenueAdapters.OkEx.Stream.TradeTest do
     :ok
   end
 
-  test "broadcasts an event when public trade is received" do
-    Events.firehose_subscribe()
-    venue_trade_id = Ecto.UUID.generate()
+  ~w(futures spot swap)
+  |> Enum.map(&"#{&1}/trade")
+  |> Enum.map(fn table ->
+    @table table
+    test "broadcasts an event when public trade is received from #{@table}" do
+      Events.firehose_subscribe()
+      venue_trade_id = Ecto.UUID.generate()
 
-    venue_trades = [
-      %{
-        "side" => "buy",
-        "trade_id" => venue_trade_id,
-        "price" => "5556.91",
-        "qty" => "5",
-        "instrument_id" => "BTC-USD-190628",
-        "timestamp" => "2019-05-06T07:19:37.496Z"
-      }
-    ]
+      venue_trades = [
+        %{
```

Comment:
Is this the same message format for spot? Specifically the `instrument_id`. I've found that the messages can often be quite different.

Reply:
Here's an sample of that unhandled message:

```json
{
  "data": {
    "msg": {
      "data": [
        {
          "instrument_id": "ADA-BTC",
          "price": "0.00000497",
          "side": "sell",
          "size": "140",
          "timestamp": "2019-09-18T04:05:18.619Z",
          "trade_id": "67497327"
        }
      ],
      "table": "spot/trade"
    },
    "received_at": "2019-09-18T04:05:18.658538Z",
    "venue_id": "okex"
  },
  "type": "Tai.StreamMessageUnhandled"
}
```
`instrument_id` looks correct, but there might be a difference in `size` -- might be an error in the test.

Reply:
I added 3 separate test for each market types